dvc/scm/git.py

```diff
@@ -89,10 +95,16 @@ def __init__(self, root_dir=os.curdir, search_parent_directories=True):
         self.ignored_paths = []
         self.files_to_track = set()
 
+        self.backends = [backend(self) for backend in self.DEFAULT_BACKENDS]
```

Comment:
Eventually `self.repo` should go away entirely (replaced by the `GitPythonBackend` in `self.backends`)